cmd/aws-k8s-agent/main.go

```diff
@@ -30,11 +30,7 @@ func main() {
 
 func _main() int {
 	//Do not add anything before initializing logger
-	logConfig := logger.Configuration{
-		LogLevel:    logger.GetLogLevel(),
-		LogLocation: logger.GetLogLocation(),
-	}
-	log := logger.New(&logConfig)
+	log := logger.Get()
```

Comment:
Since, we go through this only during initial bootup - the leaked file handles show up only if the `aws-node` pod on the node goes through a restart for some reason? I thought we've a known issue in the upstream logger lib and so curious why we are not running in to that? Would be good to expand on this in the PR description.

Reply:
>>  the leaked file handles show up only if the aws-node pod on the node goes through a restart for some reason?
No, leaked file handles show up with the long running aws-node ipamd daemon. If it is restarted, leaked file handles would disappear.
Core of the issue is that different packages used to obtain different logger instances (pointing to the same underlying file) and upon rotation, file reference would be updated in one logger but not in others, leading to leaked file handle. This change fixes the situation by ensuring that a single logger instance is ever created and the rotation logic within lumberjack ensures that rotation and file closure is properly handled. Unit-test with a single logger instance doing file rotation correctly was confirmed by using delve debugger.

Regarding the upstream issue of https://github.com/natefinch/lumberjack/issues/81, it is not directly applicable in case of ipamd as transient goroutines holding on to logger instances were not observed (using dlv).